python/lsst/pex/config/config.py

```diff
@@ -237,23 +237,26 @@ def save(self, outfile, instance):
         """
         value = self.__get__(instance)
         fullname = _joinNamePath(instance._name, self.name)
+
+        # write full documentation string as comment lines (i.e. first character is #)
+        doc = "# " + re.sub(r'\n', r'\n# ', str(self.doc))
         if isinstance(value, float) and (math.isinf(value) or math.isnan(value)):
             # non-finite numbers need special care
-            print >> outfile, "%s=float('%r')"%(fullname, value)
+            print >> outfile, "%s\n%s=float('%r')\n"%(doc, fullname, value)
```

Comment:
Interesting artifact here.  If I update `python/lsst/pex/config/config.py` to use *print-as-function*, old style *print-as-statement* syntax is no longer acceptable in config override files.  E.g., when running `processCcd.py` on `obs_subaru`, I get the following error:
```python
processCcd.py /tigress/HSC/HSC --output=/tigress/HSC/HSC/rerun/lauren/testing/ --config detection.background.useApprox=True calibrate.background.useApprox=True calibrate.detection.background.useApprox=True --id visit=904028 ccd=50 --show config=*eOut*
: Loading config overrride file '/tigress/HSC/users/lauren/LSST/obs_subaru/config/processCcd.py'
Traceback (most recent call last):
  File "/tigress/HSC/users/lauren/LSST/pipe_tasks/bin/processCcd.py", line 25, in <module>
    ProcessCcdTask.parseAndRun()
  File "/tigress/HSC/users/lauren/LSST/pipe_base/python/lsst/pipe/base/cmdLineTask.py", line 433, in parseAndRun
    parsedCmd = argumentParser.parse_args(config=config, args=args, log=log, override=cls.applyOverrides)
  File "/tigress/HSC/users/lauren/LSST/pipe_base/python/lsst/pipe/base/argumentParser.py", line 360, in parse_args
    self._applyInitialOverrides(namespace)
  File "/tigress/HSC/users/lauren/LSST/pipe_base/python/lsst/pipe/base/argumentParser.py", line 475, in _applyInitialOverrides
    namespace.config.load(filePath)
  File "/tigress/HSC/users/lauren/LSST/pex_config/python/lsst/pex/config/config.py", line 531, in load
    code = compile(f.read(), filename=filename, mode="exec")
  File "/tigress/HSC/users/lauren/LSST/obs_subaru/config/processCcd.py", line 29
    print "WARNING: Unable to use psfex: %s" % e
                                           ^
SyntaxError: invalid syntax
```

So this looks like a case where the effects of "futurizing" this one file is not isolated; I believe we would have to impose the *print-as-function* syntax for all config override file.  Thoughts?